### PR TITLE
Fight Club - Initial working code - ready for review and merge. 

### PR DESCRIPTION
**Pull request by ogmage78**
* Initial work on combat mode.   System now will move into and out of the correct combat mode depending on what items you have equipped.
* TODO: There are some edge cases I still need to implement and additional testing.   I would like to get this in so that I do not run into huge merge issues later on.
* Testing - you can log in - run @weapons after log in.   This will get you various weapons based on weapon type.   You can go into and out of combat mode
* Melee, missile - both bow and xbow, thrown, sword, UA, two handed as well as dual wield.    I also implemented if you remove your weapon while in combat mode, 
* you will drop into unarmed.   If you equip an item and you are not in peace mode, you will assume the correct combat mode.   There may be some tweaking needed 
* for changing combat mode in combat mode - I think it did some slower animation - but the basics are in and I can tweak the sequence once I talk to players or If
* I can find a pcap with that behavior.

### CODE REVIEW

File: Source/ACE/Entity/Creature.cs [hunk 1]
```diff
@@ -251,115 +251,227 @@ protected virtual void VitalTickInternal(CreatureVital vital)
...
             }
+        }
 
-            CombatMode = newCombatMode;
-            switch (CombatMode)
+        public void HandleSwitchToPeaceMode(CombatMode oldCombatMode, bool isAutonomous = false)
+        {
+            HandleUnloadMissileAmmo(oldCombatMode);
+
+            // TODO: <this is a hack for now to be removed.> Placement has an issue we have not figured out.   It has to do with animation frame. Og II
```
Comment (ddevec) [inline Source/ACE/Entity/Creature.cs]: FIXME instead of todo?

File: Source/ACE/Entity/Creature.cs [hunk 2]
```diff
@@ -251,115 +251,227 @@ protected virtual void VitalTickInternal(CreatureVital vital)
...
+                    CurrentLandblock.EnqueueBroadcast(Location, Landblock.MaxObjectRange, new GameMessageUpdatePosition(this));
+                    SetMotionState(this, mm);
+                    mm.MovementData.ForwardCommand = (ushort)MotionCommand.Reload;
+                    SetMotionState(this, mm);
+                    CurrentLandblock.EnqueueBroadcast(Location, Landblock.MaxObjectRange, new GameMessageUpdatePosition(this));
+                    combatModeChain.AddDelaySeconds(0.25);
+                    // System.Threading.Thread.Sleep(250); // used for debugging
+                    mm.MovementData.ForwardCommand = (ushort)MotionCommand.Invalid;
+                    SetMotionState(this, mm);
+                    combatModeChain.AddDelaySeconds(0.40);
```
Comment (ddevec) [inline Source/ACE/Entity/Creature.cs]: This fix probably shouldn't be part of this patch, but I really don't like these random delays -- we should put in a stub function for waiting for animation changes (which will eventually be read from our dat files)...

File: Source/ACE/Entity/Creature.cs (hunk 1 above)
Comment (ogmage78) [inline Source/ACE/Entity/Creature.cs]: I did not have that token defined.   I added it and changed it.   Updated.

File: Source/ACE/Entity/Creature.cs (hunk 2 above)
Comment (ogmage78) [inline Source/ACE/Entity/Creature.cs]: I put in FIXME's on both of these until we can get the delay times.   Without these delays, it does not function correctly.   However, just FYI these are about the correct delays - it will be interesting to see what the actual delays are.

File: Source/ACE/Entity/Creature.cs [hunk 3]
```diff
@@ -251,115 +251,227 @@ protected virtual void VitalTickInternal(CreatureVital vital)
...
+            else
+                log.InfoFormat("Changing combat mode for {0} - wielded item {1} has not be assigned a default combat style", Guid, mEquipedMelee?.Guid ?? mEquipedTwoHanded?.Guid);
+        }
+
+        public void SetCombatMode(CombatMode newCombatMode)
+        {
+            log.InfoFormat("Changing combat mode for {0} to {1}", Guid, newCombatMode);
+
+            CombatMode oldCombatMode = CombatMode;
+            CombatMode = newCombatMode;
```
Comment (ogmage78) [inline Source/ACE/Entity/Creature.cs]: Missed that completely.   I have moved this inside the action chain.   Fixed.
